<launch>
  <!-- Arguments -->
  <arg name="node_name" default="odom_to_base_link_pose_transformer"/>

  <arg name="new_poses_frame_id" default="odom"/>
  <arg name="robot_odom_frame_id" default="odom"/>
  <arg name="robot_frame_id" default="base_link_22"/>

  <arg name="poses" default="poses"/>
  <arg name="poses_odom" default="poses_base_link"/>

  <!-- Node launching -->
  <node pkg="mercator_sensor_fusion_ros_pkg" type="odom_to_base_link_pose_transformer" name="$(arg node_name)" output="screen">
    <!-- Parameters-->
    <param name="new_poses_frame_id" type="string" value="$(arg new_poses_frame_id)" />
    <param name="robot_odom_frame_id" type="string" value="$(arg robot_odom_frame_id)" />
    <param name="robot_frame_id" type="string" value="$(arg robot_frame_id)" />
    <!-- Topic remappings -->
    <remap from="poses" to="$(arg poses)" />
    <remap from="poses_odom" to="$(arg poses_odom)" />
  </node>
</launch>
